Support .NET 10

TianMengLucky opened the pull request.

monomod[bot] committed:
Add test for race condition in DynamicReferenceManager

nike4613 committed:
Fix call to original to be correct

nike4613 committed:
Merge branch 'reorganize' into ai/issue220

TheDavSmasher committed:
Created helper methods to creating a new local variable

TianMengLucky committed:
Add support for .NET 10 RC.1

TianMengLucky committed:
Update Directory.Build.props to .NET10

TianMengLucky committed:
Update packages.lock.json

nike4613 committed:
Make sure dumps target path exists so dumping to it works

nike4613 commented on build/gen-test-matrix/Models/Dotnet.cs:
```diff
@@ -111,6 +111,17 @@ internal sealed record Dotnet : Enableable
+        },
+        new()
+        {
+            Name = ".NET 10.0 RC",
```
I think I'd prefer to name this just `.NET 10.0`, and we'll just keep it updated as we support the later RCs and eventually GA.

TianMengLucky committed:
Rename ".NET 10.0 RC" to ".NET 10.0"

Rename ".NET 10.0 RC" to the suffix-free ".NET 10.0" to unify the naming.

https://github.com/MonoMod/MonoMod/pull/258#discussion_r2338876977

nike4613 committed:
Merge branch 'reorganize' into ai/issue220

TianMengLucky committed:
Update to .NET 10 RC.2

TianMengLucky committed:
Merge branch 'reorganize' into net10-rc1

TianMengLucky committed:
Update `packages.lock.json`

TianMengLucky committed:
Merge remote-tracking branch 'origin/net10-rc1' into net10-rc1

nike4613 commented:
What is suddenly including extra nullable attributes?

TianMengLucky committed:
Merge branch 'reorganize' into net10-rc1

nike4613 commented:
You didn't update iced, did you? I don't see it in the diff. What changed to trigger that?

TianMengLucky committed:
Remove nullable and System.IO.FileSystem package references

TianMengLucky committed:
Merge remote-tracking branch 'origin/net10-rc1' into net10-rc1

TianMengLucky committed:
Update `packages.lock.json`

TianMengLucky commented:
> Is the ReproducibleBuilds update maybe the problem?

No

wuke32767 committed:
also update ILHookEntry.LastContext after add ilhook

wuke32767 committed:
update test case to actually reproduce the issue

wuke32767 committed:
Merge branch 'ai/issue220' into fix-race-condition

nike4613 committed:
Merge pull request #269 from wuke32767/fix-race-condition

Fix race condition when updating executing ILHook (Referenced cell no longer exists)

Commit:
[BOT]: Bump version after

nike4613 committed:
Merge pull request #257 from TheDavSmasher/create-local-extension

Added CreateLocal helper methods

nike4613 committed:
Merge 'reorganize'

nike4613 committed:
Warning cleanup pass

nike4613 committed:
Make DetourFactory.Default and DetourFactory.Current lazy again

nike4613 committed:
Fix inner runtime poking for .NET 10

nike4613 committed:
Change MonoMod.Core's Iced reference to use Private=true

Some SDK change made PrivateAssets=all break, but Private=true work.

nike4613 committed:
Add ARM64 .NET 10 thunks
